src/index.ts

```diff
@@ -51,22 +51,43 @@ export function runInContext(
   context: Context,
   options: Partial<IOptions> = {}
 ): Promise<Result> {
-  function getFirstLine(theProgram: Program) {
-    if (theProgram.body[0] && theProgram.body[0].type === 'ExpressionStatement') {
-      const firstLineOfProgram = theProgram.body[0] as ExpressionStatement
-      const theDirective = (firstLineOfProgram as Directive).directive
-      if (theDirective !== undefined) {
-        return theDirective
-      }
+  function getFirstLine(theCode: string) {
+    function doNothing() {
+      return undefined
+    }
+
+    let theProgram
+    try {
+      theProgram = parseAt(theCode, 0)
+    } catch (error) {
```

Comment:
Why not get rid of this whole try catch block and not let parseAt throw an error in the first place?